File: src/main/java/InvertBinaryTree/GoKu-gaga/InvertBinaryTree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {TreeNode}
 */
var invertTree = function(root) {
  if (root == null || (root.left === null && root.right === null)) {
      return root;
  }     
  const left = root.left
  const right = root.right
  root.left = invertTree(right)
  root.right = invertTree(left)
  return root
};

/** 
 * 执行结果
 * 执行用时 : 80 ms, 在所有JavaScript提交中击败了90.33%的用户
 * 内存消耗 :33.6 MB, 在所有JavaScript提交中击败了65.09%的用户
*/

File: src/main/java/InvertBinaryTree/yanxianni/InvertBinaryTree.java
public class InvertBinaryTree {
	public TreeNode invertTree(TreeNode root) {
		//结束条件
		if (root == null) {
			return root;
		}
		TreeNode left = root.left;
		TreeNode right = root.right;

		root.right = invertTree(left);
		root.left = invertTree(right);
		return root;
	}

	// 定义二叉树
	public class TreeNode {
		int val;
		TreeNode left;
		TreeNode right;

		TreeNode(int x) {
			val = x;
		}
	}
}